Sources/web3swift/Utils/EIP/EIP681.swift

```diff
@@ -227,5 +347,202 @@ extension Web3 {
+                    rawValues = internalArrays
+                } else if case let .tuple(internalTypes) = type {
+                    // TODO: implement!
+                } else if case .string = type {
+                    guard let strings = splitArrayOfStrings(rawValue),
+                          (length == 0 || UInt64(strings.count) == length) else { return nil }
+                    rawValues = strings
+                } else {
+                    let rawValue = String(rawValue.dropFirst().dropLast())
+                    rawValues = rawValue.split(separator: ",").map { String($0) }
+                }
+
+                var nativeValueArray: [AnyObject] = []
+
+                for value in rawValues {
+                    let intermidiateValue = await parseFunctionArgument(type,
+                                                                        value,
+                                                                        chainID: chainID,
+                                                                        inputNumber: inputNumber)?
+                        .parameter
+                        .value
+                    if let intermidiateValue = intermidiateValue {
+                        nativeValueArray.append(intermidiateValue)
+                    }
+                }
+                nativeValue = nativeValueArray as AnyObject
+
+                guard nativeValueArray.count == rawValues.count &&
+                        (length == 0 || UInt64(rawValues.count) == length) else { return nil }
+            case let .tuple(types):
+                // TODO: implement!
+                return nil
+            default: return nil
+            }
+
+            guard let nativeValue = nativeValue else { return nil }
+            return FunctionArgument(ABI.Element.InOut(name: String(inputNumber), type: inputType),
+                                    EIP681Code.EIP681Parameter(type: inputType, value: nativeValue))
+        }
+
+        // MARK: - Parsing functions for complex data structures
+
+        /// Attempts to split given ``rawValue`` into `[String]` where each element of that array
+        /// represent a valid, stringified, array in of itself.
+        /// With an input like `"[[],[],[]]"` the output is expected to be `["[]","[]","[]"]`.
+        /// - Parameter rawValue: supposedly an array of arrays in a form `[[...],[...],...]`
+        /// - Returns: separated stringified arrays, or `nil` if separation failed.
+        private static func splitArrayOfArrays(_ rawValue: String) -> [String]? {
+            /// Dropping first and last square brackets.
+            /// That modifies the upper bound value of the first match of `squareBracketRegex`.
+            let rawValue = String(rawValue.dropFirst().dropLast())
+
+            let squareBracketRegex = try! NSRegularExpression(pattern: "(\\[*)")
+            let match = squareBracketRegex.firstMatch(in: rawValue, range: rawValue.fullNSRange)
```

Comment:
I've see that you're using regex here to make some string conversion. But isn't all input that will pass there are JSON one? If so, maybe we should consider to implement this conversions within `JSONDecode` and `RawRepresentable` rather than by parsing strings by ourselves or do I miss something?

Reply:
I'm resolving this as I've added a TODO.